extensions/data-transfer/portability-data-transfer-daybook/src/main/java/org/datatransferproject/transfer/daybook/photos/DaybookPhotosImporter.java

```diff
@@ -0,0 +1,170 @@
+ */
+
+package org.datatransferproject.transfer.daybook.photos;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.base.Preconditions;
+import com.google.common.base.Strings;
+import com.google.common.io.ByteStreams;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URL;
+import java.util.Base64;
+import java.util.UUID;
+import okhttp3.*;
+import org.datatransferproject.api.launcher.Monitor;
+import org.datatransferproject.spi.cloud.storage.TemporaryPerJobDataStore;
+import org.datatransferproject.spi.transfer.idempotentexecutor.IdempotentImportExecutor;
+import org.datatransferproject.spi.transfer.provider.ImportResult;
+import org.datatransferproject.spi.transfer.provider.Importer;
+import org.datatransferproject.transfer.JobMetadata;
+import org.datatransferproject.types.common.models.photos.PhotoAlbum;
+import org.datatransferproject.types.common.models.photos.PhotoModel;
+import org.datatransferproject.types.common.models.photos.PhotosContainerResource;
+import org.datatransferproject.types.transfer.auth.TokensAndUrlAuthData;
+
+/** Imports albums and photos to Daybook */
+public class DaybookPhotosImporter
+    implements Importer<TokensAndUrlAuthData, PhotosContainerResource> {
+
+  private final OkHttpClient client;
+  private final ObjectMapper objectMapper;
+  private final TemporaryPerJobDataStore jobStore;
+  private final Monitor monitor;
+  private final String baseUrl;
+
+  public DaybookPhotosImporter(
+      Monitor monitor,
+      OkHttpClient client,
+      ObjectMapper objectMapper,
+      TemporaryPerJobDataStore jobStore,
+      String baseUrl) {
+    this.client = client;
+    this.objectMapper = objectMapper;
+    this.jobStore = jobStore;
+    this.monitor = monitor;
+    this.baseUrl = baseUrl;
+  }
+
+  @Override
+  public ImportResult importItem(
+      UUID jobId,
+      IdempotentImportExecutor executor,
+      TokensAndUrlAuthData authData,
+      PhotosContainerResource resource)
+      throws Exception {
+    if (resource == null) {
+      // Nothing to import
+      return ImportResult.OK;
+    }
+
+    monitor.debug(() -> String.format("Number of Photos: %d", resource.getPhotos().size()));
+
+    // Import albums
+    for (PhotoAlbum album : resource.getAlbums()) {
+      executor.executeAndSwallowIOExceptions(
+          album.getId(), album.getName(), () -> importAlbum(album, authData));
+    }
+
+    // Import photos
+    for (PhotoModel photo : resource.getPhotos()) {
+      executor.executeAndSwallowIOExceptions(
+          photo.getDataId(),
+          photo.getTitle(),
+          () -> {
+            String albumId;
+            if (Strings.isNullOrEmpty(photo.getAlbumId())) {
+              albumId = null;
+            } else {
+              albumId = executor.getCachedValue(photo.getAlbumId());
+            }
+            return importPhoto(photo, jobId, authData, albumId);
+          });
+    }
+
+    // API to trigger daybook API
```

Comment:
This looks like a left over comment?

Reply:
Yes, it was.
Have removed it in the latest commit.